pkg/lwip/patches/0001-Fix-warnings.patch

```diff
@@ -1,39 +0,0 @@
-From f732fca2cd91553aa6f14ed851f84572d374aaa6 Mon Sep 17 00:00:00 2001
-From: Martine Lenders <mail@martine-lenders.eu>
-Date: Thu, 12 Nov 2015 16:36:00 +0100
-Subject: [PATCH 1/2] Fix warnings
-
----
- src/core/ipv6/nd6.c | 2 +-
- src/core/netif.c    | 2 +-
- 2 files changed, 2 insertions(+), 2 deletions(-)
-
-diff --git a/src/core/ipv6/nd6.c b/src/core/ipv6/nd6.c
-index 0b36718..b63a9b5 100644
---- a/src/core/ipv6/nd6.c
-+++ b/src/core/ipv6/nd6.c
-@@ -635,7 +635,7 @@ nd6_input(struct pbuf *p, struct netif *inp)
-           if (i >= 0) {
-             neighbor_cache[i].netif = inp;
-             MEMCPY(neighbor_cache[i].lladdr, lladdr_opt->addr, inp->hwaddr_len);
--            ip6_addr_set(&(neighbor_cache[i].next_hop_address), &tmp);
```

Comment:
I now get warnings with `ip6_addr_set` but it should be safe to ignore for this in `makefiles/lwip_ipv6.mk` as it is not NULL.

```
/home/harter/work/git/RIOT/tests/lwip/bin/pkg/native/lwip/src/include/lwip/ip6_addr.h:100:60: error: the comparison will always evaluate as ‘false’ for the address of ‘tmp’ will never be NULL [-Werror=address]
 #define ip6_addr_set(dest, src) do{(dest)->addr[0] = (src) == NULL ? 0 : (src)->addr[0]; \
                                                            ^
/home/harter/work/git/RIOT/tests/lwip/bin/pkg/native/lwip/src/core/ipv6/nd6.c:638:13: note: in expansion of macro ‘ip6_addr_set’
             ip6_addr_set(&(neighbor_cache[i].next_hop_address), &tmp);
```

Reply:
Weird... it still builds for me :-/ Maybe that patch is still required :-/

Reply:
I am with ubuntu LTS:

```
Installed toolchain versions
----------------------------
          native gcc: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.5) 5.4.0 20160609
```

Maybe now the compilers are intelligent enough to know they should not put a warning for an address check in a macro.

Reply:
I tested with Ubuntu 17.10 (`gcc (Ubuntu 7.2.0-8ubuntu3.2) 7.2.0`) and Ubuntu 16.04 (`gcc (Ubuntu 5.4.0-6ubuntu1~16.04.9) 5.4.0 20160609`.

Reply:
Ok... now I'm getting the same errors on my Ubuntu 16.04 machine... Properly didn't clean up properly :-/